CONTRIBUTING.md

```diff
@@ -100,18 +100,34 @@ checks, so please check back after a few minutes.
 
 Tests
 -----
-If your PR is a feature (ie. if it's being labelled with the 'T-Enhancement'
-label) then we require that the PR also includes tests. These need to test that
-your feature works as expected and ideally test edge cases too. For the js-sdk
-itself, your tests should generally be unit tests. matrix-react-sdk also uses
-these guidelines, so for that your tests can be unit tests using
-react-test-utils, snapshot tests or screenshot tests.
-
-We don't require tests for bug fixes (T-Defect) but strongly encourage regression
-tests for the bug itself wherever possible.
-
-In the future we may formalise this more with a minimum test coverage
-percentage for the diff.
+Your PR should include tests.
+
+For new user facing features in `matrix-react-sdk` or `element-web`, you
+must include:
+
+1. Comprehensive unit tests written in Jest. These are located in `/test`.
+2. "happy path" end-to-end tests.
+   These are located in `/test/end-to-end-tests`. Ideally, you would also
+   include tests for edge and error cases.
```

Comment:
we'll need to fix the test suite to actually run easily on all environments to make this enforceable.

Reply:
Do we have issues tracking the current problems?

Reply:
not off the top of my head, but the problems seem easily encountered. We also discussed a few of them in other channels.

It's very possible that it's just a situation of us needing to try installing them again on affected platforms and see how it goes (and report bugs).